### DISCUSSION TITLE
What is this "thin grey line" below this panel?

### DISCUSSION THREAD

**Discussion by olivia-banks** [emilk/egui#3327] [Q&A] [answered]
You can find the line directly below the menu bar. I'm unsure exactly what this is. I tried adjusting the frame stroke on both the content and the menu bar, and I don't know what else to try. The line is caused by the menu bar, as if I remove the rest of the content the line remains, and not the other way around. This is the code to create the menu bar:

```rust
TopBottomPanel::top("menu_bar").show(ctx, |ui| {
  egui::menu::bar(ui, |ui| {
    ui.menu_button("File", |ui| { /* omitted */ });
    ui.menu_button("Edit", |ui| { /* omitted */ });
    ui.menu_button("Assets", |ui| { /* omitted */ });
    ui.menu_button("Objects", |ui| { /* omitted */ });
    ui.menu_button("Components", |ui| { /* omitted */ });
    ui.menu_button("Window", |ui| { /* omitted */ });
  })
});
```

Where is this line coming from, if not the stroke?

<img width="1392" alt="Screenshot 2023-09-09 at 9 49 22 PM" src="https://github.com/emilk/egui/assets/53623746/c8ee81a4-255f-4029-9660-a15c286d6ce6">

**Comment by YgorSouza** [comment, marked as answer]
Looks like that is the separator line from the `TopBottomPanel`. There is a method to make it disappear.

```rs
egui::TopBottomPanel::top("menu_bar")
    .show_separator_line(false)
    .show(ctx, |ui| {  /* ... */ });
```

![image](https://github.com/emilk/egui/assets/43298013/d0e94e25-ca83-4c83-afee-daf55e138cc9)


And if you want to change its appearance instead of just removing it, it is controlled by `style.visuals.widgets.noninteractive.bg_stroke`.

```rs
let mut style = (*ctx.style()).clone();
style.visuals.widgets.noninteractive.bg_stroke = (1.0, Color32::RED).into();
ctx.set_style(style);
egui::TopBottomPanel::top("menu_bar")
    // .show_separator_line(false)
    .show(ctx, |ui| {  /* ... */ });
```

![image](https://github.com/emilk/egui/assets/43298013/fdaf2732-b308-46bc-bf66-e47ac41c017a)

But note that this affects a lot of other things, so you might want to change it back after the separator gets rendered.


[Screencast from 2023-09-16 09-41-34.webm](https://github.com/emilk/egui/assets/43298013/b2af813d-b53d-43a4-ab49-a2436a6d94df)
